guides/hack/02-source-code-fundamentals/13-names.md

```diff
@@ -23,15 +23,19 @@ function compute(int $val): void {
 }
 ```
 
+## The Placeholder Variable
 The name `$_`, referred to as the *placeholder variable*, is reserved for use in the
 [list intrinsic function](../expressions-and-operators/list.md) and the [foreach statement](../statements/foreach.md).
 
+## The Current Instance Variable
 The name `$this` is predefined inside any instance method or constructor when that method is called from within an object context.
 `$this` is read-only and designates the object on which the method is being called, or the object being constructed. The type of
 `$this` is [`this`](../built-in-types/this.md).
 
+## Reserved Names
 Names beginning with two underscores (__) are reserved by the Hack language.
 
+## XHP Name Constraints
 Note that [XHP classes](../XHP/introduction) have different name
 constraints. Class names may contain `:`, and must start with
```

Comment:
Not your fault. This content describes the mangling variant of xhp classes. Their source names had to start with a colon and may contain intermediary colons and minuses. These days, a colon in an xhp type name is interpreted as a backslash and minuses are disallowed.

Before `:fb:display` meant `xhp_fb__display`.
Now it means `\fb\display`.

Reply:
"must start with" is outdated and `:` is not a part of the name anymore. It is a namespace separator.

Reply:
oh, this is a good catch. I'll make a note about v3 distinctions.

Reply:
Removing the entire note on the Names page because current lib differences are more in the Namespace syntax.